src/H5Fvfd_swmr.c

```diff
@@ -1774,6 +1774,12 @@ vfd_swmr_enlarge_shadow_index(H5F_t *f)
     ret_value = shared->mdf_idx = new_mdf_idx;
     shared->mdf_idx_len         = new_mdf_idx_len;
 
+    if (shared->old_mdf_idx)
```

Comment:
This check isn't necessary.  `H5MM_xfree` checks it internally.  But it's minor.

Reply:
Fixed.